src/core/session.py

```diff
@@ -105,6 +108,18 @@ def getReadyToAddNewFeatureToObservationTable(self):
     def addNewFeatureToObservationTable(self, feature: QgsFeature):
         self._addNewFeature(feature, self._observationTable)
 
+    def addNewFeatureToGpsTable(self, longitude: float, latitude: float):
+        self._gpsTable.startEditing()
+
+        feature = QgsFeature(QgsVectorLayerUtils.createFeature(self._gpsTable))
+        layerPoint = QgsPointXY(longitude, latitude)
+        feature.setGeometry(QgsGeometry.fromPointXY(layerPoint))
+        feature.setAttribute(
+            SammoDataBase.GPS_TIME_FIELD_NAME, self.nowToString()
```

Comment:
The datetime comes from the GPS frame so ti should be a parameter of `addNewFeatureToGpsTable`